File: apps/api/app/schemas/control_point.py
from pydantic import BaseModel, ConfigDict


class ControlPointCreate(BaseModel):
    device_id: int
    control_type: str
    source_domain_id: int
    destination_domain_id: int
    supports_simulation: bool
    priority: int


class ControlPointRead(ControlPointCreate):
    model_config = ConfigDict(from_attributes=True)

    id: int

File: packages/contracts/src/control-point.ts
import { z } from "zod";

export const controlPointSchema = z.object({
  id: z.string(),
  deviceId: z.string(),
  controlType: z.enum(["boundary_block", "interzone_isolation", "ops_access", "audit", "logging"]),
  sourceDomainId: z.string(),
  destinationDomainId: z.string(),
  supportsSimulation: z.boolean().default(true),
  priority: z.number().int().default(100)
});

export type ControlPoint = z.infer<typeof controlPointSchema>;
